packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -132,12 +140,35 @@ export class SnapInterfaceController extends BaseController<
     super({
       messenger,
       metadata: {
-        interfaces: { persist: false, anonymous: false },
+        interfaces: {
+          persist: (interfaces: Record<string, StoredInterface>) => {
+            return Object.entries(interfaces).reduce<
+              Record<string, StoredInterface>
+            >((persistedInterfaces, [id, snapInterface]) => {
+              switch (snapInterface.contentType) {
+                case ContentType.Notification:
+                  persistedInterfaces[id] = snapInterface;
+                  return persistedInterfaces;
+                default:
+                  return persistedInterfaces;
+              }
+            }, {});
+          },
+          anonymous: false,
+        },
       },
       name: controllerName,
       state: { interfaces: {}, ...state },
     });
 
+    this._onNotificationsListUpdated =
```

Comment:
I don't think we need this if we make the function hash private and bind it inside the subscribe call. See:  https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/insights/SnapInsightsController.ts#L116